File: lib/models/product.dart
import 'dart:convert';

class Product {
  final String name;
  final String description;
  final double price;
  final double quantity;
  final String category;
  final List<String> images;
  final String? id;

  Product(
      {required this.name,
      required this.description,
      required this.price,
      required this.quantity,
      required this.category,
      required this.images,
      this.id,
      });


  Map<String, dynamic> toMap() {
    return {
      'name': name,
      'description': description,
      'price': price,
      'quantity': quantity,
      'category': category,
      'images': images,
      'id': id,

    };
  }

  factory Product.fromMap(Map<String, dynamic> map) {
    return Product(
      name: map['name'] ?? '',
      description: map['description'] ?? '',
      price: map['price'] ?? 0.0,
      quantity: map['quantity'] ?? 0,
      category: map['category'] ?? '',
      images: List<String>.from(map['images']),
      id: map['_id'] ?? '',

    );
  }

  String toJson() => json.encode(toMap());

  factory Product.fromJson(String source) =>Product.fromMap(json.decode(source));
}

File: server/models/product.js
const mongoose = require('mongoose');
const productSchema = mongoose.Schema({
    name:{
        type:String,
        required:true,
        trim:true,
    },
    price:{
        type:Number,
        required:true,
    },
    description:{
        type:String,
        required:true,
        trim:true,
    },
    image:[{
        type:String,
        required:true,
    },
    ],
    category:{
        type:String,
        required:true,
        
    },
    quantity:{
        type:Number,
        required:true,
    },
    sold:{
        type:Number,
        default:0,
    }

    

});

const Product =mongoose.model('Product',productSchema);
module.exports = Product;